src/suffixtree.jl

```diff
@@ -0,0 +1,112 @@
+using Base.Test
+import Base.insert!
+
+abstract SuffixTrie
+
+type SuffixTrieNode <: SuffixTrie
+  value
+  children::Array{SuffixTrieNode}
```

Comment:
Without a dimension `Array{SuffixTrieNode}` is actually an abstract type, and in general, it's almost always better to use concrete types for members of types.  So this should either be `Vector{SuffixTrieNode}` or `Array{SuffixTrieNode,1}` (which mean exactly the same thing).